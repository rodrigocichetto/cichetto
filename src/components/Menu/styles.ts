import styled, { css } from 'styled-components';

export const Wrapper = styled.nav`
  ${({ theme }) => css`
    position: absolute;
    box-shadow: ${theme.shadow};
    padding: ${theme.spacings.xsmall};
    background: ${theme.colors.menu};
    border-radius: ${theme.spacings.xxlarge};
    bottom: ${theme.spacings.small};
    left: 50%;
    transform: translateX(-50%);
  `}
`;
