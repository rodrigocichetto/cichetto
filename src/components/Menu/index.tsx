import { useCallback } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';

import { Icon } from 'components';
import { AvailableIcons } from 'components/Icon';

import * as S from './styles';

export type MenuPage = {
  path: string;
  icon: AvailableIcons;
  label: string;
};

export const PAGES: MenuPage[] = [
  {
    path: '/xp',
    icon: 'xp',
    label: 'ExperiÃªncia'
  },
  {
    path: '/',
    icon: 'user',
    label: 'Home'
  },
  {
    path: '/gamer',
    icon: 'gamer',
    label: 'Gamer'
  }
];

const Menu = () => {
  const router = useRouter();

  const isActive = useCallback((path: string) => router.pathname === path, [
    router.pathname
  ]);

  return (
    <S.Wrapper>
      {PAGES.map(page => (
        <S.LinkWrapper key={page.path} active={isActive(page.path)}>
          <Link href={page.path} passHref>
            <a>
              <Icon size="xxlarge" icon={page.icon} />
            </a>
          </Link>
        </S.LinkWrapper>
      ))}
    </S.Wrapper>
  );
};

export default Menu;
