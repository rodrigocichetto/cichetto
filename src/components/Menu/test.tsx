import { renderWithTheme } from 'utils/tests/helpers';

import Menu from '.';

describe('<Menu />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Menu />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
