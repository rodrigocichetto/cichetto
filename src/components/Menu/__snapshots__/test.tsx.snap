// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render correctly 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c2 svg {
  height: 2.8rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  box-shadow: 0 0 5rem rgba(0,0,0,0.1);
  padding: 1.6rem;
  background: #343844;
  border-radius: 5.6rem;
  bottom: 2.4rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 20;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0 > * {
  margin: 0 1.6rem;
}

.c1 {
  cursor: pointer;
  opacity: 0.5;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c1:hover {
  -webkit-transform: translateY(-0.8rem);
  -ms-transform: translateY(-0.8rem);
  transform: translateY(-0.8rem);
}

@media (min-width:768px) {
  .c0 {
    position: absolute;
  }

  .c0:hover {
    bottom: 3.2rem;
  }
}

@media (max-width:768px) {
  .c0 svg {
    height: 2.0rem;
  }
}

<nav
  class="c0"
>
  <div
    class="c1"
  >
    <a
      href="/"
    >
      <div
        class="c2"
        color="text"
      >
        <svg
          aria-hidden="true"
          aria-label="user"
          class="c3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
            fill="currentColor"
          />
        </svg>
      </div>
    </a>
  </div>
  <div
    class="c1"
  >
    <a
      href="/gamer"
    >
      <div
        class="c2"
        color="text"
      >
        <svg
          aria-hidden="true"
          aria-label="gamer"
          class="c3"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M467.51 248.83c-18.4-83.18-45.69-136.24-89.43-149.17A91.5 91.5 0 00352 96c-26.89 0-48.11 16-96 16s-69.15-16-96-16a99.09 99.09 0 00-27.2 3.66C89 112.59 61.94 165.7 43.33 248.83c-19 84.91-15.56 152 21.58 164.88 26 9 49.25-9.61 71.27-37 25-31.2 55.79-40.8 119.82-40.8s93.62 9.6 118.66 40.8c22 27.41 46.11 45.79 71.42 37.16 41.02-14.01 40.44-79.13 21.43-165.04z"
            fill="none"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="32"
          />
          <circle
            cx="292"
            cy="224"
            r="20"
          />
          <path
            d="M336 288a20 20 0 1120-19.95A20 20 0 01336 288z"
          />
          <circle
            cx="336"
            cy="180"
            r="20"
          />
          <circle
            cx="380"
            cy="224"
            r="20"
          />
          <path
            d="M160 176v96m48-48h-96"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="32"
          />
        </svg>
      </div>
    </a>
  </div>
</nav>
`;
